retis/src/process/cli/pcap.rs

```diff
@@ -200,42 +200,46 @@ impl<'a, W: Write> EventParser<'a, W> {
 #[derive(Parser, Debug, Default)]
 #[command(name = "pcap")]
 pub(crate) struct Pcap {
+    #[command(flatten)]
+    cmd: PcapCmd,
     #[arg(
         short,
         long,
-        help = "Filter events from this probe. Probes should follow the [TYPE:]TARGET pattern.
-See `retis collect --help` for more details on the probe format."
+        conflicts_with = "list_probes",
+        help = "Write the generated PCAP output to a file rather than stdout"
     )]
-    pub(super) probe: String,
+    pub(super) out: Option<PathBuf>,
+    #[arg(default_value = "retis.data", help = "File from which to read events")]
+    pub(super) input: PathBuf,
+}
+#[derive(Args, Debug, Default)]
+#[group(required = true, multiple = false)]
+pub(crate) struct PcapCmd {
+    #[arg(short, long, help = "List probes that are available in the input file")]
+    pub(super) list_probes: bool,
     #[arg(
         short,
         long,
-        help = "Write the generated PCAP output to a file rather than stdio"
+        help = "Filter events from this probe. Probes should follow the [TYPE:]TARGET pattern.
+See `retis collect --help` for more details on the probe format"
     )]
-    pub(super) out: Option<PathBuf>,
-    #[arg(default_value = "retis.data", help = "File from which to read events")]
-    pub(super) input: PathBuf,
+    pub(super) probe: Option<String>,
 }
 
 impl SubCommandParserRunner for Pcap {
     fn run(&mut self, _: &MainConfig) -> Result<()> {
-        let (probe_type, target) = parse_cli_probe(&self.probe)?;
+        if self.cmd.list_probes {
+            let probes = list_probes(self.input.as_path())?;
+            probes.iter().for_each(|p| println!("{p}"));
+            return Ok(());
+        }
+        // With derive API for clap, we can check for the presence of:
+        // -l | -p <> [-o <>]
+        // But we still have to catch the case where 0 arguments were provided.
```

Comment:
This can be removed now. You can replace it with a comment on the following`unwrap` call.

Other than that the PR looks good.